src/super_gradients/training/sg_model/sg_model.py

```diff
@@ -102,6 +105,20 @@ class EvaluationType(str, Enum):
     VALIDATION = 'VALIDATION'
 
 
+from dataclasses import dataclass
+
+
+@dataclass
+class EpochStats:
+    current: float = None
+    previous: float = None
+    best: float = None
+    change_from_previous: float = None
+    change_from_best: float = None
+    is_better_than_previous: bool = None
+    is_best_epoch: bool = None
```

Comment:
This could be use to standardize the format of the dictionaries "self.train_epoch_stats" and "self.train_epoch_stats" (instantiated line 895)

Pros:
- forces the format
- less repetition when creating new instances (just call EpochStats() instead of initializing all the fields 1 by 1)

Cons:
- Different coding style (Introduces dataclasses which are not used in other places)
- "Only" used in 3 places, and not necessary to have understandable code.